plugin/pkg/scheduler/algorithm/priorities/priorities.go

```diff
@@ -89,7 +89,42 @@ func LeastRequestedPriority(pod *api.Pod, podLister algorithm.PodLister, minionL
 
 	list := algorithm.HostPriorityList{}
 	for _, node := range nodes.Items {
-		list = append(list, calculateOccupancy(pod, node, podsToMachines[node.Name]))
+		list = append(list, calculateResourceOccupancy(pod, node, podsToMachines[node.Name]))
+	}
+	return list, nil
+}
+
+func min(l, r int64) (m int64) {
+	m = r
+	if l < r {
+		m = l
+	}
+	return m
+}
+
+// See comment for DumbSpreadingPriority()
+const dumbSpreadingDenominator int64 = 10
+
+// DumbSpreadingPriority is a priority function that favors nodes with fewer pods.
+// It works like LeastRequestedPeriority but instead of using 10 * percentage of machine free by resource,
+// it uses 10 * percentage of machine free by pod, with "percentage of machine free by pod" claculated as
+// (dumbSpreadingDenominator - number of pods already on the node + 1) / dumbSpreadingDenominator.
+// dumbSpreadingDenominator serves like the machine capacity in LeasRequestedPriority but is chosen
+// so that we equate one pod with a reasonable amount of resources when we combine all the scores together.
```

Comment:
I hate discrete functions... I also mostly agree with @davidopp, but I also believe that @bprashanth observation about maxing out score pretty quickly is justified. IIUC you just zero the score with 10th pod running and scheduler won't distinguish Node with 10 running pods from one with 99 running ones (assuming a number of 0-limit ones and the same resource requests for the rest). In continuous world we could just apply some log or other cot function to it and be happy, for discrete ones it's harder, but still I'd consider doing something similar. We'll lose some resolution at the bottom, but we'll gain some differentiation between loaded nodes.

I guess we should flatten at 30 Pods, as this is officially supported number, but I believe we should distinguish 10 pods/node from 25. E.g. something like 10-[2*log_2(x+1)]. Which has some "bad" properties (big drop at the beginning: 1->9, 2->7, 3->6), but in general I believe values it gives are sane for < 63 pods/node..